pkg/config/config.go

```diff
@@ -646,6 +649,21 @@ type Configuration struct {
 	root *yaml.Node
 }
 
+// CPE stores values used to produce a CPE to describe the package, suitable for
+// matching against NVD records.
+//
+// For Melange, the "part" attribute should always be interpreted as "a" (for
+// "application").
+type CPE struct {
+	Vendor  string
+	Product string
+}
+
+// String returns a CPE string for the package.
+func (c CPE) String() string {
+	return fmt.Sprintf("cpe:2.3:a:%s:%s:*:*:*:*:*:*:*:*", c.Vendor, c.Product)
```

Comment:
Is there more we might want to be able to fill out? I never remember CPE semantics, but I think it's also got version range info, and we have version info. Can we use it in any way?

Reply:
Yeah definitely. It can do versions (not ranges), so we can plug in the `.Package.Version` field from the `Configuration` struct. Nothing's calling this `String()` method yet, but if someone does, they'd get the benefit of that.

Same goes for other fields. We could add them all now, if we want. I was curious if it'd be better to wait and see which we end up needing (I'd bet money we don't need them all), but no opinion here.

Reply:
I'm not against adding those two, but I will say:

1. I almost never see those set. I'm not sure people know how to use them. And for OS, more often I see the CVE record using boolean groups of multiple CPEs, rather than trying to convey the OS in a single CPE.
2. From the perspective of scanning, which is the big motivation here, by the time we're looking at a package, we already know the architecture and that it's Linux, so we don't _really_ need the package itself to tell us that.